### PR TITLE
Merge Monitoring tool changes to MME

### PR DESCRIPTION
**Pull request by badhrinathpa**

### CODE REVIEW

File: include/common/monitor_message.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2003-2018, Great Software Laboratory Pvt. Ltd.
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
```
Comment (thakurajayL) [inline include/common/monitor_message.h]: update ONF copyright here.

File: include/common/monitor_message.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+#define IMSI_STR_LEN 16
+/*AIR  - request. Struct is same for ULR*/
+struct monitor_imsi_req {
+    int  req_type;
+	char imsi[IMSI_STR_LEN];
+};
+
+/*AIA  - response*/
```
Comment (thakurajayL) [inline include/common/monitor_message.h]: typo ?

File: include/common/monitor_message.h [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+#ifndef __MONITOR_MSG_H_
+#define __MONITOR_MSG_H_
+
+#include "ue_table.h"
+#include "s1ap_structs.h"
+#include "s11_structs.h"
+
+/*Information needed for AIR*/
```
Comment (thakurajayL) [inline include/common/monitor_message.h]: typo ?

File: include/common/monitor_message.h [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+struct monitor_resp_msg {
+	enum e_BUF_HDR hdr;
+	int result;
+	union resp_data {
+		struct monitor_imsi_rsp mia;
+	}data;
+};
+
+#define MONITOR_RCV_BUF_SIZE  128
```
Comment (thakurajayL) [inline include/common/monitor_message.h]: Any relation with number of subscribers in the MME ?

File: include/common/unix_conn.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2003-2018, Great Software Laboratory Pvt. Ltd.
+ * Copyright (c) 2017 Intel Corporation
```
Comment (thakurajayL) [inline include/common/unix_conn.h]: copyright to ONF

File: pyserver/monitor_client.py [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+    def default(self, o):
+        if type(o) is bytes:
+            return o.decode("ISO-8859-1")
+        return super(MyEncoder, self).default(o)
+
+app = Flask(__name__)
+
+@app.route("/")
+
+def helloWorld():
```
Comment (thakurajayL) [inline pyserver/monitor_client.py]: may be we should get rid of this or put some other banner ?

File: pyserver/monitor_client.py [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+            (bytes_to_mnc(parsed_data.tai.plmn_id.idx))}
+        account.append(data)
+        print(json.dumps(payload_in, cls=MyEncoder))
+        output = (json.dumps(payload_in, cls=MyEncoder))
+        output1 = json.dumps(parsed_data.__dict__)
+
+        return jsonify(account)
+
+
+if __name__ == '__main__':
```
Comment (thakurajayL) [inline pyserver/monitor_client.py:211]: what is this port ?

File: src/mme-app/handlers/attach_initue.c [hunk 8]
```diff
@@ -162,10 +164,13 @@ stage1_processing(struct s6a_Q_msg *s6a_req, struct attachReqRej_info *s1ap_rej,
       unsigned char mnc1 = plmn_byte2 >> 4;  // mnc1
       unsigned char mcc3  = plmn_byte2 & 0xf; //mcc3
       // First byte we are not changing             mcc2 mcc1
-      plmn_byte2 = (mnc3 << 4) | mcc3; // 2nd byte on NAS - mnc3 mcc3
-      plmn_byte3 = (mnc2 << 4) | mnc1; // 3rd byte on NAS - <mnc2 mnc1>
-      ue_info->tai.plmn_id.idx[1] = plmn_byte2;
-      ue_info->tai.plmn_id.idx[2] = plmn_byte3;
+      if(mnc1 != 0x0F)
```
Comment (thakurajayL) [inline src/mme-app/handlers/attach_initue.c:167]: I will review this separately. Requires more time to see if we are aligned with all 4 encoder/decoder - s11, s6a, s1ap & NAS. For now please make sure that 3 digit mcc/mnc is not broken.

File: pyserver/monitor_client.py (hunk 7 above)
Comment (badhrinathpa) [reply pyserver/monitor_client.py:211]: its the port the flask server runs on.

File: pyserver/monitor_client.py (hunk 6 above)
Comment (badhrinathpa) [reply pyserver/monitor_client.py]: done

File: include/common/unix_conn.h (hunk 5 above)
Comment (badhrinathpa) [reply include/common/unix_conn.h]: done

File: include/common/monitor_message.h (hunk 4 above)
Comment (badhrinathpa) [reply include/common/monitor_message.h]: Not used. Removing

File: include/common/monitor_message.h (hunk 3 above)
Comment (badhrinathpa) [reply include/common/monitor_message.h]: done

File: include/common/monitor_message.h (hunk 2 above)
Comment (badhrinathpa) [reply include/common/monitor_message.h]: done

File: include/common/monitor_message.h (hunk 1 above)
Comment (badhrinathpa) [reply include/common/monitor_message.h]: done

**Pull request merged**